omero/developers/json-api.txt

```diff
@@ -0,0 +1,526 @@
+      "url:servers": "http://<server>/api/v0/servers/",
+      "url:images": "http://<server>/api/v0/m/images/"
+    }
+
+
+List available OMERO servers
+----------------------------
+
+Your API may allow you to connect to several different OMERO servers.
+
+::
+
+    GET     /api/v0/servers/
+
+
+**Response**
+
+::
+
+
+
+    {
+      "data": [
+        {
+          "host": "<server>",
+          "server": "omero",
+          "id": 1,
+          "port": 4064
+        }
+      ]
+    }
+
+
+Get CSRF token
+--------------
+
+In order to prevent CSRF attacks, CSRF tokens are required for any
+POST, PUT and DELETE requests. You will need to obtain a CSRF token
+for your session and use it for all subsequent requests in that session.
+You can obtain this from the ``csrftoken``
+cookie of any request or from the following URL:
+
+::
+
+    GET     /api/v0/token/
+
+
+**Response**
+
+::
+
+    {
+      "data": "eNoVq528bOqlhQqbCzKuviODTRX3PUO2"
+    }
+
+
+Login
+-----
+
+You can login to create an OMERO session. You must also include the CSRF token,
+either in the POST parameters as 'csrfmiddlewaretoken' or in the session
+header as 'X-CSRFToken'.
+
+The EventContext for this session will be returned to you.
+
+::
+
+    POST    /api/v0/login/
+
+
+**Parameters**
+
+::
+
+    Name                  Type        Description
+    ------------------------------------------------------------------
+    server                Number      ID of the server
+    username              String      User's username
+    password              String      User's password
+    csrfmiddlewaretoken   String      CSRF token (can be provied in header)
+
+
+**Response**
+
+::
+
+    {
+      "eventContext": {
+        "userName": "ben",
+        "eventId": -1,
+        "sessionUuid": "0b30ee4a-c0b2-4b0f-9c61-f48b31bcad8c",
+        "eventType": "User",
+        "userId": 3,
+        "sessionId": 171319,
+        "groupName": "Nevis Lab",
+        "isAdmin": False,
+        "memberOfGroups": [5, 1, 4],
+        "leaderOfGroups": [],
+        "groupId": 5
+      },
+      "success": true
+    }
+
+
+Image containers
+----------------
+
+OMERO organises images in two types of many-to-many hierarchies:
+``screen/plate/[run]/well/image`` for HCS data and ``project/dataset/image``
+for other data. Plates, Datasets and Images can also be ``Orphaned`` if not
+contained within any parent container.
+
+
+**Parameters**
+
+These query parameters are used by many queries below:
+
+::
+
+    Name        Type        Description
+    ------------------------------------------------------------------
+    offset      Number      Pagination offset. The default is 0.
+
+    limit       Number      The size of each page. The default is 200.
+
+    normalize   Boolean     Place Experimenters and Groups into top-level lists instead
+                            of nesting within objects
+    childCount  Boolean     Use ?childCount=true to include an omero:childCount attribute
```

Comment:
I wonder about trying some backticks around the literal code bits for the description in this kind of text